plugins/browser-plugin-media-tracking/src/player.ts

```diff
@@ -0,0 +1,174 @@
+import { v4 as uuidv4 } from 'uuid';
+import {
+  startMediaTracking,
+  MediaEventType,
+  trackMediaPlay,
+  trackMediaPause,
+  trackMediaEnd,
+  trackMediaSeekEnd,
+  trackMediaPlaybackRateChange,
+  trackMediaVolumeChange,
+  trackMediaFullscreenChange,
+  trackMediaPictureInPictureChange,
+  trackMediaBufferStart,
+  trackMediaBufferEnd,
+  trackMediaError,
+  updateMediaTracking,
+  trackMediaReady,
+} from '@snowplow/browser-plugin-media';
+import { MediaPlayerUpdate, MediaType } from '@snowplow/browser-plugin-media/src/types';
+import { ElementConfig } from './config';
+import { getDuration, isFullScreen, parseVolume } from './helperFunctions';
+import { buildHTMLMediaElementEntity, buildHTMLVideoElementEntity } from './entities';
+import { LOG, SelfDescribingJson } from '@snowplow/tracker-core';
+
+type CommonMediaPlayerUpdate = Omit<MediaPlayerUpdate, 'fullscreen' | 'pictureInPicture' | 'quality'>;
+type VideoMediaPlayerUpdate = Pick<MediaPlayerUpdate, 'fullscreen' | 'pictureInPicture'>;
+
+function updatePlayer(el: HTMLMediaElement): MediaPlayerUpdate {
+  const common: CommonMediaPlayerUpdate = {
+    currentTime: el.currentTime || 0,
+    duration: getDuration(el),
+    ended: el.ended,
+    livestream: el.duration === Infinity,
+    loop: el.loop,
+    mediaType: el.tagName === 'VIDEO' ? MediaType.Video : MediaType.Audio,
+    muted: el.muted,
+    paused: el.paused,
+    volume: parseVolume(el.volume),
+    playbackRate: el.playbackRate,
+  };
+
+  const video: VideoMediaPlayerUpdate =
+    el instanceof HTMLVideoElement
+      ? {
+          fullscreen: isFullScreen(el),
+          pictureInPicture: document.pictureInPictureElement === el,
+        }
+      : {};
+
+  return { ...common, ...video };
+}
+
+function htmlContext(el: HTMLMediaElement): (() => SelfDescribingJson)[] {
+  const context = [() => buildHTMLMediaElementEntity(el)];
+
+  if (el instanceof HTMLVideoElement) {
+    context.push(() => buildHTMLVideoElementEntity(el));
+  }
+
+  return context;
+}
+
+export function setUpListeners(config: ElementConfig) {
+  const { video } = config;
+  const id = uuidv4();
```

Comment:
We should let users provide their own `id` (the media session id) similar like we do in the Vimeo plugin. Not sure we can make it generate a uuid by default because then the user is unable to stop the media tracking?